### PR TITLE
Refine functions and add parsing test to set and zset 

### LINKED ISSUE #1469: Refine functions and add parsing tests to `Set`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

Many commands in the current [`cmd_set`](https://github.com/apache/incubator-kvrocks/blob/b9cfbead58f68aa914bb35360f82c3f8e0ec8c90/src/commands/cmd_set.cc) file miss parsing error tests(It is the same as `zset`.) and misses some necessary checks.

For example, `syntax: SPOP key [count]` from https://redis.io/commands/spop/.
https://github.com/apache/incubator-kvrocks/blob/b9cfbead58f68aa914bb35360f82c3f8e0ec8c90/src/commands/cmd_set.cc#L142-L155

### Solution

we can add the error check `"value is not an integer or out of range"`. And then we add some go tests to match parsing errors. 

You can just add some codes for one class or more classes(in `set` or `zset`).

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zncleon**
- Refine functions
`spop`
`srandmember`
`zpop`

- Add parsing test
`spop`
`srandmember`
`zadd`
`zincby`
`zlexcount`
`zpop`
`zunionstore`

In addition, `srandmember` not consider the case of a negative count and return an empty list or set，should we give it a definite action?

Close #1469

### CODE REVIEW

File: src/commands/cmd_zset.cc [hunk 1]
```diff
@@ -223,7 +223,9 @@ class CommandZPop : public Commander {
   explicit CommandZPop(bool min) : min_(min) {}
 
   Status Parse(const std::vector<std::string> &args) override {
-    if (args.size() > 2) {
+    if (args.size() > 3) {
+      return {Status::RedisParseErr, errWrongNumOfArguments};
+    } else if (args.size() == 3) {
```
Comment (git-hulk) [inline src/commands/cmd_zset.cc]:
```suggestion
    }

    if (args.size() == 3) {
```

**Review by git-hulk — APPROVED** [review]
LGTM

Thanks for your contribution.

**Comment by git-hulk** [comment]
Hi @zncleon 

>In addition, srandmember not consider the case of a negative count and return an empty list or set，should we give it a definite action?

In this case, we can behave consistently with the Redis.

**Review by PragmaTwice — APPROVED** [review]

**Review by torwig — APPROVED** [review]

**Comment by infdahai** [comment]
a nice contribution!

**Comment by zncleon** [comment]
ok，@git-hulk
> In this case, we can behave consistently with the Redis.

now `srandmember` only returns always first N members if not changed.
I will implement its(as well as `zrandmember` #1459) random algorithm according to Redis.

**Comment by infdahai** [comment]
you can do that in another PR. By the way, if you want to write that function, you should think about the member/membersource function in zset(by referring to member func in set).

**Comment by infdahai** [comment]
yeah, you can do this(or give a new PR about this special problem).

**Comment by git-hulk** [comment]
> ok，@git-hulk
> 
> > In this case, we can behave consistently with the Redis.
> 
> now `srandmember` only returns always first N members if not changed. I will implement its(as well as `zrandmember` #1459) random algorithm according to Redis.

@zncleon Cool, can submit an issue to track this.

**Comment by git-hulk** [comment]
Thanks all, mergin...

**Pull request merged**